kind.sh

```diff
@@ -29,55 +29,65 @@ fi
 
 if [[ "$CMD" == "create" ]]; then
     CONFIG=kind-config.yaml
-    cat > $CONFIG << EOF
-# three node (two workers) cluster config
+
+    # Only write the config if it's not present; this allows customization 
+    if ! [[ -f "$CONFIG" ]]; then
+
+      # Rabbit & WLM System Local Controllers (SLC)
+      SLCCONFIG=$(cat << EOF
+
+  kubeadmConfigPatches:
+  - |
+    kind: JoinConfiguration
+    nodeRegistration:
+      kubeletExtraArgs:
+        node-labels: cray.nnf.manager=true,cray.wlm.manager=true
+EOF
+)
+
+      # Rabbit taints/labels, plus some host mounts for data movement
+      RABBITCONFIG=$(cat << EOF
+
+  extraMounts:
+    - hostPath: /tmp/nnf
+      containerPath: /nnf
+      propagation: None
+  kubeadmConfigPatches:
+  - |
+    kind: JoinConfiguration
+    nodeRegistration:
+      taints:
+      - key: cray.nnf.node
+        value: "true"
+        effect: NoSchedule
+      kubeletExtraArgs:
+        node-labels: "cray.nnf.node=true"
+EOF
+)
+
+      cat > $CONFIG <<EOF
 kind: Cluster
 apiVersion: kind.x-k8s.io/v1alpha4
 networking:
   apiServerAddress: "127.0.0.1"
 nodes:
 - role: control-plane
-- role: worker
-  extraMounts:
-  - hostPath: /tmp/nnf
-    containerPath: /nnf
-    propagation: None
-- role: worker
-  extraMounts:
-  - hostPath: /tmp/nnf
-    containerPath: /nnf
-    propagation: None
+- role: worker $SLCCONFIG
+- role: worker $RABBITCONFIG
+- role: worker $RABBITCONFIG
 EOF
 
+    fi
+
+    # create a file for data movement
     if [ ! -f /tmp/nnf/file.in ]; then
         mkdir -p /tmp/nnf && dd if=/dev/zero of=/tmp/nnf/file.in bs=128 count=0 seek=$((1024 * 1024))
     fi
```

Comment:
Can we do data movement on kind?

Reply:
Yes. Although you still need to hack the command so it's copying from /nnf/file.in to /nnf/file.out.